src/components/overlay/overlay.scss

```diff
@@ -0,0 +1,193 @@
+// travel the shadow boundary are defined through this mixin
+@include sbb.host-component-properties;
+
+:host {
+  @include sbb.scrollbar-variables;
+
+  --sbb-overlay-padding-inline: var(--sbb-spacing-fixed-5x);
+  --sbb-overlay-color: var(--sbb-color-black);
+  --sbb-overlay-background-color: var(--sbb-color-milk);
+  --sbb-overlay-width: 100%;
+  --sbb-overlay-height: 100%;
+  --sbb-overlay-inset: 0 auto auto 0;
+  --sbb-overlay-padding-block: var(--sbb-spacing-responsive-s);
+  --sbb-overlay-animation-duration: var(--sbb-animation-duration-6x);
+  --sbb-overlay-animation-easing: ease;
+  --sbb-overlay-pointer-events: none;
+  --sbb-overlay-content-transition: transform var(--sbb-overlay-animation-duration)
+    var(--sbb-overlay-animation-easing);
+
+  position: fixed;
+  inset: var(--sbb-overlay-inset);
+  z-index: var(--sbb-overlay-z-index, var(--sbb-overlay-z-index));
+
+  @include sbb.mq($from: micro) {
+    --sbb-overlay-padding-inline: var(--sbb-spacing-fixed-6x);
+  }
+
+  @include sbb.mq($from: small) {
+    --sbb-overlay-padding-inline: var(--sbb-spacing-fixed-12x);
+  }
+
+  @include sbb.mq($from: medium) {
+    --sbb-overlay-padding-inline: var(--sbb-spacing-responsive-s);
+  }
+}
+
+:host(:is([data-state='opened'], [data-state='opening'])) {
+  --sbb-overlay-pointer-events: all;
+}
+
+:host([negative]) {
+  @include sbb.scrollbar-variables--color-negative;
+
+  --sbb-focus-outline-color: var(--sbb-focus-outline-color-dark);
+  --sbb-overlay-color: var(--sbb-color-white);
+  --sbb-overlay-background-color: var(--sbb-color-midnight);
+}
+
+:host([disable-animation]) {
+  --sbb-overlay-animation-duration: 0.1ms;
+}
+
+:host(:not([data-state='closed'])) {
+  --sbb-overlay-inset: 0;
+}
+
+.sbb-overlay__container {
+  pointer-events: var(--sbb-overlay-pointer-events);
+  display: flex;
+  align-items: center;
+  position: fixed;
+  inset: var(--sbb-overlay-inset);
+}
+
+.sbb-overlay {
+  display: none;
+  position: absolute;
+  inset-inline: 0;
+  margin: auto;
+  padding: 0;
+  border: none;
+  width: var(--sbb-overlay-width);
+  height: var(--sbb-overlay-height);
+  overflow: auto;
+  color: var(--sbb-overlay-color);
+  background-color: var(--sbb-overlay-background-color);
+
+  :host([data-state]:not([data-state='closed'])) & {
+    display: block;
+
+    animation: {
+      name: open;
+      duration: var(--sbb-overlay-animation-duration);
+      timing-function: ease;
+    }
+  }
+
+  :host([data-state][data-state='closing']) & {
+    pointer-events: none;
+    animation-name: close;
+  }
+
+  @include sbb.if-forced-colors {
+    outline: var(--sbb-border-width-1x) solid CanvasText;
+  }
+
+  @include sbb.mq($from: medium) {
+    overflow: hidden;
+    height: fit-content;
+  }
+}
+
+.sbb-overlay__wrapper {
+  display: flex;
+  flex-direction: column;
+  width: var(--sbb-overlay-width);
+  height: var(--sbb-overlay-height);
+  outline: none;
+  position: fixed;
+  word-break: break-word;
+
+  :host(:not([expanded])) & {
```

Comment:
Why are you applying page spacing to wrapper and header?
I'm not sure whether we should just use a container in the template (like in Figma) and we have delegated the responsibility to the container.